samples/cpp/speech_sample/speech_sample.hpp

```diff
@@ -23,10 +23,10 @@ static const char model_message[] = "Required. Path to an .xml file with a train
 
 /// @brief message for assigning calculation to device
 static const char target_device_message[] =
-    "Optional. Specify a target device to infer on. CPU, GPU, MYRIAD, GNA_AUTO, GNA_HW, "
+    "Optional. Specify a target device to infer on. CPU, GPU, MYRIAD, VPUX, GNA_AUTO, GNA_HW, "
```

Comment:
@Wovchena @nkogteva do we know why we have a hardcoded list of devices? Other samples print devices via core.available devices

Reply:
I don't know. Maybe it comes from older times when there was no such function in IE.

Reply:
Actually there is a call to `showAvailableDevices()` to form the list dynamically: https://github.com/openvinotoolkit/openvino/blob/cd9602547f984d4c1e6ef0a5ed140ccae2802166/samples/cpp/speech_sample/speech_sample.hpp#L240

Static version of device list `supportedDevices` is only used to `throw std::logic_error("Specified device is not supported.");`. I think `supportedDevices` can be removed as well as the list from help message.

Don't forget to modify python help as well and update Running section of READMEs

Reply:
@ilya-lavrenov @Wovchena It is kept for historical reasons. BTW there are no such devices like GNA_HW, GNA_SW_EXACT and so on. There is GNA device plus hardware settings. So it is kind of simlpification whcih allows user to choose configuration by one command line parameter.

Reply:
@AsyaPronina please, address comments

Reply:
Hello Ilya(@ilya-lavrenov), Natalia(@nkogteva) and  Vladimir(@Wovchena)! Thank you a lot for review!

As I've got the result of the discussion, I should save `supportedDevices` because of convenience for user to choose GNA configuration via single parameter, but  I have to update python sample and README-s. Am I right?
If yes, could you please check the updates?

Reply:
> Hello Ilya(@ilya-lavrenov), Natalia(@nkogteva) and Vladimir(@Wovchena)! Thank you a lot for review!
> 
> As I've got the result of the discussion, I should save `supportedDevices` because of convenience for user to choose GNA configuration via single parameter, but I have to update python sample and README-s. Am I right? If yes, could you please check the updates?

LGTM

Reply:
Thank you a lot!